src/librustc_borrowck/borrowck/mir/elaborate_drops.rs

```diff
@@ -586,6 +586,11 @@ impl<'b, 'tcx> ElaborateDropsCtxt<'b, 'tcx> {
                             // drop elaboration should handle that by itself
                             continue
                         }
+                        TerminatorKind::Resume => {
+                            // We can replace resumes with gotos
+                            // jumping to a canonical resume.
```

Comment:
Nope, still necessary. Okay so I'll figure out what I did here that caused this to be exposed, and then improve the explanation in the comment.